### PR TITLE
Upgrade to 0.107.0

### PR DESCRIPTION
**Pull request by iblancasa**

### CODE REVIEW

File: _build/build.log [hunk 1]
```diff
@@ -1,8 +1,9 @@
...
-2024-08-13T18:51:07.479+0200	INFO	internal/command.go:161	Using config file	{"path": "manifest.yaml"}
-2024-08-13T18:51:07.489+0200	INFO	builder/config.go:142	Using go	{"go-executable": "/home/ploffay/bin/go/bin/go"}
-2024-08-13T18:51:07.495+0200	INFO	builder/main.go:100	Sources created	{"path": "./_build"}
-2024-08-13T18:51:07.704+0200	INFO	builder/main.go:191	Getting go modules
-2024-08-13T18:51:07.766+0200	INFO	builder/main.go:111	Compiling
-2024-08-13T18:51:12.779+0200	INFO	builder/main.go:130	Compiled	{"binary": "./_build/otelcol"}
+2024-08-14T10:03:30.744+0200	INFO	internal/command.go:125	OpenTelemetry Collector Builder	{"version": "0.106.1", "date": "2024-07-30T18:19:16Z"}
+2024-08-14T10:03:30.746+0200	INFO	internal/command.go:161	Using config file	{"path": "manifest.yaml"}
+2024-08-14T10:03:30.758+0200	INFO	builder/config.go:142	Using go	{"go-executable": "/home/iblancasa/go/bin/go"}
+2024-08-14T10:03:30.758+0200	INFO	builder/main.go:77	You're building a distribution with non-aligned version of the builder. Compilation may fail due to API changes. Please upgrade your builder or API	{"builder-version": "0.106.1"}
```
Comment (frzifus) [inline _build/build.log:5]: Should we increase the builder version too?

Comment (iblancasa) [reply _build/build.log:5]: No binary was released for 0.107.0
https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.107.0

Comment (pavolloffay) [reply _build/build.log:5]: it has been moved here https://github.com/open-telemetry/opentelemetry-collector-releases/releases/tag/cmd%2Fbuilder%2Fv0.107.0

Comment (pavolloffay) [reply _build/build.log:5]: https://github.com/open-telemetry/opentelemetry-collector-releases/pull/608

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**